### PR TITLE
[IOTDB-4045] Optimize DataNode query relevant interfaces

### PR DESCRIPTION
**Pull request by 23931017wu**
### remove the input parameter of getRegisteredDataNodes and getRegisteredDataNodeInfoList since we always query the registered DataNodes at the same time, just like query registered ConfigNodes.

### CODE REVIEW

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 1]
```diff
@@ -364,8 +371,13 @@ private TSStatus transferLeader(
   }
 
   public List<TSStatus> merge(TMergeReq req) {
-    Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap;
+    if (req.dataNodeId == -1) {
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:375]:
```suggestion
    if (req.dataNodeId != -1) {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 2]
```diff
@@ -375,8 +387,13 @@ public List<TSStatus> merge(TMergeReq req) {
   }
 
   public List<TSStatus> flush(TFlushReq req) {
-    Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap;
+    if (req.dataNodeId == -1) {
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:391]:
```suggestion
    if (req.dataNodeId != -1) {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 3]
```diff
@@ -386,8 +403,13 @@ public List<TSStatus> flush(TFlushReq req) {
   }
 
   public List<TSStatus> clearCache(TClearCacheReq req) {
-    Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+    Map<Integer, TDataNodeLocation> dataNodeLocationMap;
+    if (req.dataNodeId == -1) {
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:407]:
```suggestion
    if (req.dataNodeId != -1) {
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 4]
```diff
@@ -204,17 +204,24 @@ public int getTotalCpuCoreCount() {
...
   }
 
-  public Map<Integer, TDataNodeLocation> getRegisteredDataNodeLocations(int dataNodeId) {
+  public Map<Integer, TDataNodeLocation> getRegisteredDataNodeLocationById(int dataNodeId) {
+    Map<Integer, TDataNodeLocation> dataNodeLocations = new ConcurrentHashMap<>();
+    dataNodeLocations.put(
+        nodeInfo.getRegisteredDataNodeById(dataNodeId).getLocation().getDataNodeId(),
+        nodeInfo.getRegisteredDataNodeById(dataNodeId).getLocation());
+    return dataNodeLocations;
+  }
```
Comment (CRZbulabula) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:219]: An interface that returns a map with only one element looks a little strange?

**Review by CRZbulabula — CHANGES_REQUESTED** [review]
PTAL

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java (hunk 4 above)
Comment (23931017wu) [reply confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java]: > An interface that returns a map with only one element looks a little strange?
Okay, I've modified it ~

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 5]
```diff
@@ -365,7 +365,13 @@ private TSStatus transferLeader(
 
   public List<TSStatus> merge(TMergeReq req) {
     Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+        configManager.getNodeManager().getRegisteredDataNodeLocations();
+    if (req.dataNodeId != -1) {
+      dataNodeLocationMap =
+          dataNodeLocationMap.entrySet().stream()
+              .filter((e) -> 1 == e.getKey())
```
Comment (RYH61) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:372]:
```suggestion
              .filter((e) -> req.dataNodeId == e.getKey())
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 6]
```diff
@@ -376,7 +382,13 @@ public List<TSStatus> merge(TMergeReq req) {
 
   public List<TSStatus> flush(TFlushReq req) {
     Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+        configManager.getNodeManager().getRegisteredDataNodeLocations();
+    if (req.dataNodeId != -1) {
+      dataNodeLocationMap =
+          dataNodeLocationMap.entrySet().stream()
+              .filter((e) -> 1 == e.getKey())
```
Comment (RYH61) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java:389]:
```suggestion
              .filter((e) -> req.dataNodeId == e.getKey())
```

File: confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java [hunk 7]
```diff
@@ -387,7 +399,13 @@ public List<TSStatus> flush(TFlushReq req) {
 
   public List<TSStatus> clearCache(TClearCacheReq req) {
     Map<Integer, TDataNodeLocation> dataNodeLocationMap =
-        configManager.getNodeManager().getRegisteredDataNodeLocations(req.dataNodeId);
+        configManager.getNodeManager().getRegisteredDataNodeLocations();
+    if (req.dataNodeId != -1) {
+      dataNodeLocationMap =
+          dataNodeLocationMap.entrySet().stream()
+              .filter((e) -> 1 == e.getKey())
```
Comment (RYH61) [inline confignode/src/main/java/org/apache/iotdb/confignode/manager/NodeManager.java]:
```suggestion
              .filter((e) -> req.dataNodeId == e.getKey())
```

**Review by CRZbulabula — APPROVED** [review]
LGTM～

**Review by RYH61 — APPROVED** [review]
LGTM～

**Review by neuyilan — APPROVED** [review]
LGTM

**Pull request merged**